File: Components/WinForms/VB/DiagramMainDemo/DiagramTutorialControl.vb
Imports System.ComponentModel
Imports System.Linq
Imports System.Windows.Forms
Imports DevExpress.DXperience.Demos
Imports DevExpress.Utils
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraDiagram.Utils

Namespace DevExpress.XtraDiagram.Demos
    Partial Public Class DiagramTutorialControl
        Inherits DiagramTutorialControlBase

        Protected Overrides ReadOnly Property Ribbon() As RibbonControl
            Get
                Return If(containerControl Is Nothing, Nothing, containerControl.Controls.OfType(Of RibbonControl)().FirstOrDefault())
            End Get
        End Property

        Public Sub New()
            InitializeComponent()
            Diagram.CreateDocking()
            Diagram.CreateRibbon()
        End Sub
    End Class
End Namespace

File: Components/WinForms/CS/DiagramMainDemo/DiagramTutorialControl.cs
using System.ComponentModel;
using System.Linq;
using System.Windows.Forms;
using DevExpress.DXperience.Demos;
using DevExpress.Utils;
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraDiagram.Utils;

namespace DevExpress.XtraDiagram.Demos {
    public partial class DiagramTutorialControl : DiagramTutorialControlBase {
        protected override RibbonControl Ribbon {
            get { return containerControl == null ? null : containerControl.Controls.OfType<RibbonControl>().FirstOrDefault(); }
        }

        public DiagramTutorialControl() {
            InitializeComponent();
            Diagram.CreateDocking();
            Diagram.CreateRibbon();
        }
    }
}
